server_management/management/commands/pushdb.py

```diff
@@ -14,81 +14,85 @@ def handle(self, *args, **options):
-                        name=remote['database']['name'],
-                        owner=remote['database']['user'],
-                    ),
-                },
-                {
-                    'title': 'Ensure user has access to the database',
-                    'command': 'sudo su - postgres -c "psql {name} -c \'GRANT ALL ON DATABASE {name} TO {owner}\'"'.format(
-                        name=remote['database']['name'],
-                        owner=remote['database']['user'],
-                    ),
-                },
-                {
-                    'title': 'Ensure user does not have unnecessary privileges',
-                    'command': 'sudo su - postgres -c "psql {name} -c \'ALTER USER {owner} WITH NOSUPERUSER NOCREATEDB\'"'.format(
-                        name=remote['database']['name'],
-                        owner=remote['database']['user'],
-                    ),
-                },
-                {
-                    'title': 'Start Supervisor tasks',
-                    'command': 'sudo supervisorctl start all',
-                },
-            ]
-
-            run_tasks(env, db_tasks)
-
-            # Import the database file
-            # sudo("su - {name} -c 'psql -q {name} < /tmp/{name}.sql > /dev/null 2>&1'".format(
-
-            with settings(sudo_user=remote['database']['user']):
-                sudo('psql {name} < /tmp/{name}.sql'.format(
-                    name=remote['database']['name']
-                ))
-
-            # Remove the database file
-            run('rm /tmp/{}.sql'.format(
-                remote['database']['name']
-            ))
-
-            # Remove the SQL file from the host
-            local('rm ~/{}.sql'.format(
-                config['local']['database']['name']
-            ))
+        config, connection = load_config(options.get('remote', ''), debug=options.get('debug', False))
+        remote = config['remotes'][config['remote_name']]
+
+        title_print('Backing up database', 'task')
+        perform_backup(config, connection)
+        title_print('Backing up database', 'succeeded')
+
+        # Create a final dump of the database. We use Invoke's run() here as Postgres' pg_dump won't work since postgres won't be in $PATH for connection.local()
+        local('pg_dump {name} -cOx -U {user} -f ~/{name}.sql --clean'.format(
+            name=config['local']['database']['name'],
+            user=os.getlogin()
+        ))
+
+        # Push the database from earlier up to the server
+        connection.local('scp{}~/{}.sql {}@{}:/{}'.format(
+            f' -i {connection.connect_kwargs["key_filename"]} ' if connection.connect_kwargs.get('key_filename') else ' ',
+            config['local']['database']['name'],
+            connection.user,
+            remote['server']['ip'],
+            os.path.join(
+                'tmp',
```

Comment:
This really should be `/tmp/` for the universally-used temporary directory on Unix systems (otherwise it'll use `tmp` relative to the current directory).